src/actions/account/index.ts

```diff
@@ -77,12 +71,40 @@ export const checkIdentityExists: ThunkAction = async (
   // TODO: rework the seed param on lib, currently cleartext seed is being passed around. Bad.
   const userVault = JolocomLib.KeyProvider.fromSeed(
     Buffer.from(decryptedSeed, 'hex'),
-    password,
+    encryptionPass,
   )
 
-  await backendMiddleware.setIdentityWallet(userVault, password)
-  const identityWallet = backendMiddleware.identityWallet
-  dispatch(setDid(identityWallet.identity.did))
+  const userPubKey = userVault.getPublicKey({
+    derivationPath: KeyTypes.jolocomIdentityKey,
+    encryptionPass,
+  })
+
+  const didDocument = await storageLib.get.didDoc(publicKeyToDID(userPubKey))
+
+  if (didDocument) {
+    const identity = Identity.fromDidDocument({ didDocument })
+
+    // TODO Simplify constructor
+    backendMiddleware.identityWallet = new IdentityWallet({
```

Comment:
I think this section should better go inside the method currently called `authenticateAndSetIdentityWallet`. Or maybe call it `createIdentityWallet` like I mentioned in another comment, and it could decide between either creating a wallet object like this or using `registry.authenticate`